### ISSUE TITLE
no error when subscripting `AbstractContextManager` when targeting python 3.8

### ISSUE THREAD

**Issue by DetachHead** [microsoft/pyright#6996]
**Describe the bug**
when targeting python 3.8, pyright correctly warns when you use newer typing features in a position where they get evaluated at runtime and would cause a runtime error (eg. subscripting `list`), however it does not warn when subscripting `AbstractContextManager`

**Code or Screenshots**
```python
from contextlib import AbstractContextManager

bar: list[int] # error: Subscript for class "list" will generate runtime exception; enclose type annotation in quotes
foo: AbstractContextManager[object] # no error
```
[playground](https://pyright-play.net/?pythonVersion=3.8&strict=true&code=GYJw9gtgBAxmB2AXApgD0QGwJYCMpYgAcwREoBBHAZ0RAEMZEBhBFdAWTvjoHNkQAsAChhOOiABcUbDQDaWJAF0oAYij9wkqAGUArtRggshMsBKwMdKlSgAiGYltQA7lgwYofePzoooIXSQCZHVUGGQTLAQAbnV4GAwwKhDEAE9CEK54MERfKPh8AoBHXRzkKmEzMClKGnpGFiQ0RE5uPhBZMBwAK2RGZTVs9RBNIA)

**Comment by erictraut** [comment]
Yeah, pyright doesn't claim to report all such issues. There are nearly an infinite number of bugs and odd behaviors in various versions of the runtime. Pyright reports common ones, but you'll need to look to other tools if you're interested in more comprehensive checks. Perhaps ruff has some linter rules that could help you out.
